includes/class-wc-gateway-amazon-payments-advanced-subscriptions.php

```diff
@@ -318,27 +318,32 @@ public function recurring_checkout_session_update( $payload, $checkout_session_i
 			'amount'    => null,
 		);
 
+		$recurring_total = 0;
+		foreach ( WC()->cart->recurring_carts as $recurring_cart ) {
+			$recurring_total += $recurring_cart->get_total( 'edit' );
+		}
+
+		$recurring_total = wc_format_decimal( $recurring_total, '' );
+
 		if ( 1 === $subscriptions_in_cart ) {
-			$recurring_total  =  0;
-			foreach ( WC()->cart->recurring_carts as $recurring_cart ) {
-				$recurring_total += $recurring_cart->get_total( 'edit' );
-			}
 			$payload['recurringMetadata']['amount'] = array(
 				'amount'       => $recurring_total,
 				'currencyCode' => wc_apa_get_order_prop( $order, 'order_currency' ),
 			);
 		}
 
-		if ( "0.00" === WC()->cart->get_total( 'edit' ) ) {
-			$payload['paymentDetails']['paymentIntent'] = "Confirm";
+		if ( 0 <= WC()->cart->get_total( 'edit' ) ) {
```

Comment:
when I debugged this the value was a string, this conditional works without the (float) ?   and also why is 0 <= total and not 0>=total?

Reply:
Something like this?

```suggestion
		if ( 0 >= (float) WC()->cart->get_total( 'edit' ) ) {
```